src/hotspot/share/runtime/arguments.cpp

```diff
@@ -501,7 +501,19 @@ static SpecialFlag const special_jvm_flags[] = {
   { "RequireSharedSpaces",          JDK_Version::jdk(18), JDK_Version::jdk(19), JDK_Version::undefined() },
   { "UseSharedSpaces",              JDK_Version::jdk(18), JDK_Version::jdk(19), JDK_Version::undefined() },
   { "RegisterFinalizersAtInit",     JDK_Version::jdk(22), JDK_Version::jdk(23), JDK_Version::jdk(24) },
-
+#if defined(X86)
+  { "UseRTMLocking",                JDK_Version::jdk(23), JDK_Version::jdk(24), JDK_Version::jdk(25) },
```

Comment:
Why do you include experimental options?

Only `UseRTMLocking`, `UseRTMDeopt`, and `RTMRetryCount` are product.

```
src/hotspot//cpu/x86/globals_x86.hpp:  product(bool, UseRTMLocking, false,                                       \
src/hotspot//cpu/x86/globals_x86.hpp:  product(bool, UseRTMForStackLocks, false, EXPERIMENTAL,                   \
src/hotspot//cpu/x86/globals_x86.hpp:  product(bool, UseRTMDeopt, false,                                         \
src/hotspot//cpu/x86/globals_x86.hpp:  product(int, RTMRetryCount, 5,                                            \
src/hotspot//cpu/x86/globals_x86.hpp:  product(int, RTMSpinLoopCount, 100, EXPERIMENTAL,                         \
src/hotspot//cpu/x86/globals_x86.hpp:  product(int, RTMAbortThreshold, 1000, EXPERIMENTAL,                       \
src/hotspot//cpu/x86/globals_x86.hpp:  product(int, RTMLockingThreshold, 10000, EXPERIMENTAL,                    \
src/hotspot//cpu/x86/globals_x86.hpp:  product(int, RTMAbortRatio, 50, EXPERIMENTAL,                             \
src/hotspot//cpu/x86/globals_x86.hpp:  product(int, RTMTotalCountIncrRate, 64, EXPERIMENTAL,                     \
src/hotspot//cpu/x86/globals_x86.hpp:  product(intx, RTMLockingCalculationDelay, 0, EXPERIMENTAL,                \
src/hotspot//cpu/x86/globals_x86.hpp:  product(bool, UseRTMXendForLockBusy, true, EXPERIMENTAL,                  \
src/hotspot//share/opto/c2_globals.hpp:  product(bool, PrintPreciseRTMLockingStatistics, false, DIAGNOSTIC,        \
```

Reply:
I listed all RTM flags to get VM's deprecation message for all of them.

Reply:
But I missed `PrintPreciseRTMLockingStatistics`. I will added it too.

Reply:
CSR may need only to list product flags but I decided to list them all in CSR too.

Reply:
> I listed all RTM flags to get VM's deprecation message for all of them.

Any particular point in doing so? IMO it looks excessive. Moreover, all flags imply `UseRTMLocking`. 

Also, I don't see any other non-product flags in `special_jvm_flags` table you modify.

Reply:
Some RTM flags are guarded by `UseRTMLocking` and some are not: [java.cpp#L267](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/java.cpp#L267)

Reply:
I want this PR to be clear message that we deprecate all RTM related flags with goal to remove all of them and RTM code from HotSpot. 
Actually `java.1` man page listed `RTMAbortRatio` even so it is experimental. It listed product flags as well.
Which reminds me that I need to update `java.1` man page to move these flags to `DEPRECATED JAVA OPTIONS` section.

Reply:
Okay, lets ask main expert @dholmes-ora what we should do: list only product flags in this PR and CSR or listed in `java.1` or all?

Reply:
@dholmes-ora told me to list only product flags in PR and CSR. And update man page.

Reply:
Done.